packages/web/components/complex/all-pools-table.tsx

```diff
@@ -0,0 +1,707 @@
+      ]
+    );
+
+    const tvlFilteredPools = useMemo(
+      () =>
+        allPoolsWithMetrics.filter((p) =>
+          p.liquidity.toDec().gte(new Dec(TVL_FILTER_THRESHOLD))
+        ),
+      [allPoolsWithMetrics]
+    );
+
+    const poolFilteredPools = useMemo(
+      () =>
+        tvlFilteredPools.filter((p) => {
+          if (poolFilter) {
+            return p.pool.type === poolFilter;
+          }
+          return true;
+        }),
+      [poolFilter, tvlFilteredPools]
+    );
+
+    const incentiveFilteredPools = useMemo(
+      () =>
+        poolFilteredPools.filter((p) => {
+          if (incentiveFilter === "superfluid") {
+            return queriesOsmosis.querySuperfluidPools.isSuperfluidPool(
+              p.pool.id
+            );
+          }
+          if (incentiveFilter === "internal") {
+            return queriesOsmosis.queryIncentivizedPools.isIncentivized(
+              p.pool.id
+            );
+          }
+          if (incentiveFilter === "external") {
+            const gauges = queryActiveGauges.getExternalGaugesForPool(
+              p.pool.id
+            );
+            return gauges && gauges.length > 0;
+          }
+          return true;
+        }),
+      [
+        incentiveFilter,
+        poolFilteredPools,
+        queriesOsmosis.queryIncentivizedPools,
+        queriesOsmosis.querySuperfluidPools,
+        queryActiveGauges,
+      ]
+    );
+
+    const [query, _setQuery, filteredPools] = useFilteredData(
+      incentiveFilteredPools,
+      useMemo(
+        () => [
+          "pool.id",
+          "poolName",
+          "networkNames",
+          "pool.poolAssets.amount.currency.originCurrency.pegMechanism",
+        ],
+        []
+      )
+    );
+    const setQuery = useCallback(
+      (search: string) => {
+        if (search !== "" && !fetchedRemainingPoolsRef.current) {
+          queriesOsmosis.queryGammPools.fetchRemainingPools();
+          fetchedRemainingPoolsRef.current = true;
+        }
+        setSorting([]);
+        _setQuery(search);
+      },
+      [_setQuery, queriesOsmosis.queryGammPools]
+    );
+
+    const [cellGroupEventEmitter] = useState(() => new EventEmitter());
+    const tableData: Pool[] = useMemo(
+      () =>
+        filteredPools.map((poolWithMetrics) => {
+          const poolId = poolWithMetrics.pool.id;
+          const poolAssets = poolWithMetrics.pool.poolAssets.map(
+            (poolAsset) => ({
+              coinImageUrl: poolAsset.amount.currency.coinImageUrl,
+              coinDenom: poolAsset.amount.currency.coinDenom,
+            })
+          );
+
+          const pool: Pool = [
+            {
+              poolId,
+              poolAssets,
+              stableswapPool: poolWithMetrics.pool.type === "stable",
+            },
+            { value: poolWithMetrics.liquidity },
+            {
+              value: poolWithMetrics.volume24h,
+              isLoading: !queriesExternalStore.queryGammPoolFeeMetrics.response,
```

Comment:
`queriesExternalStore.queryGammPoolFeeMetrics` has an `isFetching` member that would be more idiomatic to use for figuring out if something is fetching.